src/backend/drm/compositor/mod.rs

```diff
@@ -0,0 +1,2756 @@
+            .field("damage", &self.damage)
+            .field("states", &self.states)
+            .field("primary_element", &self.primary_element)
+            .field("overlay_elements", &self.overlay_elements)
+            .field("cursor_element", &self.cursor_element)
+            .finish()
+    }
+}
+
+render_elements! {
+    /// Render elements for the [`RenderFrameResult`]
+    pub FrameRenderElements<'a, R, E>;
+    /// A texture representing the contents of the slot used
+    /// for rendering the primary plane
+    Slot=TextureRenderElement<<R as Renderer>::TextureId>,
+    /// A single element that was used for direct scan-out
+    Element=&'a E
+}
+
+impl<'a, R, E> std::fmt::Debug for FrameRenderElements<'a, R, E>
+where
+    R: Renderer,
+    <R as Renderer>::TextureId: std::fmt::Debug,
+    E: RenderElement<R> + std::fmt::Debug,
+{
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        match self {
+            Self::Slot(arg0) => f.debug_tuple("Slot").field(arg0).finish(),
+            Self::Element(arg0) => f.debug_tuple("Element").field(arg0).finish(),
+            Self::_GenericCatcher(arg0) => f.debug_tuple("_GenericCatcher").field(arg0).finish(),
+        }
+    }
+}
+
+impl<'a, B, E, R> AsRenderElements<R> for RenderFrameResult<'a, B, E>
+where
+    B: Buffer + AsDmabuf,
+    <B as AsDmabuf>::Error: std::fmt::Debug,
+    E: Element + RenderElement<R>,
+    R: Renderer + ImportDma,
+    <R as Renderer>::TextureId: 'static,
+{
+    type RenderElement = FrameRenderElements<'a, R, E>;
+
+    fn render_elements<C: From<Self::RenderElement>>(
+        &self,
+        renderer: &mut R,
+        _location: Point<i32, Physical>,
+        _scale: Scale<f64>,
+    ) -> Vec<C> {
+        self.filtered_render_elements(renderer, [])
+    }
+}
+
+/// Specialized renderer utilizing drm planes with support for direct scan-out
+#[derive(Debug)]
+pub struct DrmCompositor<A, F, U>
+where
+    A: Allocator,
+    F: ExportFramebuffer<A::Buffer>,
+    <F as ExportFramebuffer<A::Buffer>>::Framebuffer: std::fmt::Debug + 'static,
+{
+    output: Output,
+    surface: Arc<DrmSurface>,
+    planes: Planes,
+    damage_tracked_renderer: DamageTrackedRenderer,
+    primary_plane_element_id: Id,
+    primary_plane_damage_tracker: DamageTracker<i32, BufferCoords>,
+
+    framebuffer_exporter: F,
+
+    current_frame: Frame<A, F>,
+    pending_frame: Option<(Frame<A, F>, U)>,
+    queued_frame: Option<(Frame<A, F>, U)>,
+    next_frame: Option<Frame<A, F>>,
+
+    swapchain: Swapchain<A>,
+
+    cursor_size: Size<i32, Physical>,
+    cursor_swapchain: Swapchain<A>,
+    cursor_damage_tracked_renderer: DamageTrackedRenderer,
+
+    last_scale: Scale<f64>,
+    last_transform: Transform,
+
+    element_states: IndexMap<Id, ElementFramebufferCache<<F as ExportFramebuffer<A::Buffer>>::Framebuffer>>,
+
+    debug_flags: DebugFlags,
+}
+
+const SUPPORTED_FORMATS: &[DrmFourcc] = &[DrmFourcc::Argb8888, DrmFourcc::Xrgb8888];
```

Comment:
Is this something we would have to have as a parameter when creating a DrmCompositor?

Reply:
I copied that part over from `GbmBufferedSurface` but forgot the comment. 

Updated to include the comment which explains the reason behind defining that formats